### PR TITLE
Show warning if nosto module output is disabled

### LINKED ISSUE #278: Add a warning banner if the Nosto module output is disabled

### PR DESCRIPTION
**Pull request by chenliangde**
Closes #278

### CODE REVIEW

File: app/code/community/Nosto/Tagging/Model/Observer.php [hunk 1]
```diff
@@ -265,4 +280,34 @@ public function syncNostoAccount(/** @noinspection PhpUnusedParameterInspection
...
+        Varien_Event_Observer $observer)
+    {
+        if (self::$hasModuleOutputChecked) {
+            return;
+        }
+        self::$hasModuleOutputChecked = true;
+
+        /** @var Nosto_Tagging_Helper_Data $helper */
+        $helper = Mage::helper('nosto_tagging');
+        if (!$helper->isModuleEnabled()) {
```
Comment (mridang) [inline app/code/community/Nosto/Tagging/Model/Observer.php]: What is the different between this and the `Mage::getStoreConfig(self::MODULE_OUTPUT_DISABLE_SETTING_PATH)`?

Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Model/Observer.php]: They are not the same thing. This is about module disabling: https://github.com/Nosto/nosto-magento/blob/develop/app/etc/modules/Nosto_Tagging.xml#L31

Comment (mridang) [inline app/code/community/Nosto/Tagging/Model/Observer.php]: @chenliangde Is the module-output is disabled then how is the iframe being displayed? Isn't that theoretically output as well? Are you implying that disable-module-output only affects output on the front-end of the site?

File: app/code/community/Nosto/Tagging/etc/config.xml [hunk 2]
```diff
@@ -120,6 +120,15 @@
                     </nosto_tagging_custom>
                 </observers>
             </admin_system_config_changed_section_nosto_tagging>
+            <adminhtml_block_html_before>
```
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/etc/config.xml]: We really shouldn't use / observe this layout handle. This event / handle is fired 20-30 times on every page load in store admin. We must find a handle that is fired only once

File: app/code/community/Nosto/Tagging/Model/Observer.php [hunk 3]
```diff
@@ -39,11 +39,26 @@
 class Nosto_Tagging_Model_Observer
 {
     /**
+     * URL path of advanced setting
+     */
+    const ADVANCED_CONFIG_URL_PATH = 'adminhtml/system_config/edit/section/advanced';
```
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Model/Observer.php]: I would add this to the data helper where all the other similar constants are defined

File: app/code/community/Nosto/Tagging/Model/Observer.php [hunk 4]
```diff
@@ -265,4 +280,34 @@ public function syncNostoAccount(/** @noinspection PhpUnusedParameterInspection
 
         return $this;
     }
+
+    /**
+     * check if the nosto module output is disabled.
+     * @param Varien_Event_Observer $observer
+     */
+    public function checkIfModuleOutputDisabled(/** @noinspection PhpUnusedParameterInspection */ // @codingStandardsIgnoreLine
```
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Model/Observer.php]: I would add this to the data helper where all the other similar methods are defined

File: app/code/community/Nosto/Tagging/Model/Observer.php (hunk 1 above)
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Model/Observer.php]: If the nosto module output is disabled, then the iframe will not be loaded. At least it is not loaded on magento 1.14.

**Comment by chenliangde** [comment]
If the nosto module output has been disabled, then the nosto iframe will not be loaded.  That makes this requirement invalid.

**Pull request closed without merging**